spec/decorators/prosecution_case_decorator_spec.rb

```diff
@@ -43,50 +49,132 @@
-    let(:hearing2_days) { ['2021-01-20T16:00:00.000Z'] }
-    let(:hearing3_days) { ['2021-01-18T11:00:00.000Z'] }
+    context 'when sort_colum is date and direction is desc' do
+      before do
+        allow(decorator).to receive(:column).and_return('date')
+        allow(decorator).to receive(:direction).and_return('desc')
+      end
+
+      let(:expected_days) do
+        ['2021-01-20T16:00:00.000Z'.to_datetime,
+         '2021-01-20T10:45:00.000Z'.to_datetime,
+         '2021-01-19T10:45:00.000Z'.to_datetime,
+         '2021-01-18T11:00:00.000Z'.to_datetime]
+      end
 
-    let(:expected_days) do
-      ['2021-01-18T11:00:00.000Z'.to_datetime,
-       '2021-01-19T10:45:00.000Z'.to_datetime,
-       '2021-01-20T10:45:00.000Z'.to_datetime,
-       '2021-01-20T16:00:00.000Z'.to_datetime]
+      it { expect(call.map(&:day)).to eql(expected_days) }
     end
 
-    it { is_expected.to be_instance_of(Enumerator) }
-    it { is_expected.to all(be_instance_of(HearingDecorator)) }
-    it { expect(call.map(&:day)).to contain_exactly(*expected_days) }
+    context 'when sort_colum is type and direction is asc' do
+      before do
+        allow(decorator).to receive(:column).and_return('type')
+        allow(decorator).to receive(:direction).and_return('asc')
+      end
+
+      let(:expected_days) do
+        ['2021-01-20T16:00:00.000Z'.to_datetime,
+         '2021-01-18T11:00:00.000Z'.to_datetime,
+         '2021-01-19T10:45:00.000Z'.to_datetime,
+         '2021-01-20T10:45:00.000Z'.to_datetime]
+      end
+
+      it { expect(call.map(&:day)).to eql(expected_days) }
+    end
+
+    context 'when sort_colum is type and direction is desc' do
+      before do
+        allow(decorator).to receive(:column).and_return('type')
+        allow(decorator).to receive(:direction).and_return('desc')
+      end
+
+      let(:expected_days) do
+        ['2021-01-19T10:45:00.000Z'.to_datetime,
+         '2021-01-20T10:45:00.000Z'.to_datetime,
+         '2021-01-18T11:00:00.000Z'.to_datetime,
+         '2021-01-20T16:00:00.000Z'.to_datetime]
+      end
+
+      it { expect(call.map(&:day)).to eql(expected_days) }
+    end
+
+    context 'when sort_colum is provider and direction is asc' do
+      before do
+        allow(decorator).to receive(:column).and_return('provider')
+        allow(decorator).to receive(:direction).and_return('asc')
+        allow(decorated_hearing1).to receive(:provider_list).and_return('Jammy Dodger (Junior)')
+        allow(decorated_hearing2).to receive(:provider_list).and_return('Custard Cream (QC)')
+        allow(decorated_hearing3).to receive(:provider_list).and_return('Choc Digestive (QC)<br>Hob Nob (QC)')
+      end
+
+      let(:expected_days) do
+        ['2021-01-18T11:00:00.000Z'.to_datetime,
+         '2021-01-20T16:00:00.000Z'.to_datetime,
+         '2021-01-19T10:45:00.000Z'.to_datetime,
+         '2021-01-20T10:45:00.000Z'.to_datetime]
+      end
+
+      it { expect(call.map(&:day)).to eql(expected_days) }
+    end
```

Comment:
[ditto](https://github.com/ministryofjustice/laa-court-data-ui/pull/485/files#r605680744) but for provider

Reply:
👍 I have amended this to test expect(call.map(&:provider_list)) to attempt to make this clearer